test/extensions/common/aws/signer_impl_test.cc

```diff
@@ -41,6 +41,27 @@ class SignerImplTest : public testing::Test {
     message_->body() = std::make_unique<Buffer::OwnedImpl>(body);
   }
 
+  void expectSignHeaders(absl::string_view service_name, absl::string_view signature,
+                         absl::string_view payload) {
+    auto* credentials_provider = new NiceMock<MockCredentialsProvider>();
```

Comment:
nit: why use new and then create a shared_ptr later?
This should should be:
`CredentialsProviderSharedPtr credentials_provider = std::make_shared<NiceMock<MockCredentialsProvider>>();`

Reply:
maybe no member named 'gmock_getCredentials' in 'Envoy::Extensions::Common::Aws::CredentialsProvider'